components/higher-order/with-notices/index.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * External dependencies
+ */
+import uuid from 'uuid/v4';
+
+/**
+ * WordPress dependencies
+ */
+import { Component, createHigherOrderComponent } from '@wordpress/element';
+
+/**
+ * Override the default edit UI to include notices if supported.
+ *
+ * @param  {function|Component} BlockEdit Original component.
+ * @return {Component}                    Wrapped component.
+ */
+export default createHigherOrderComponent( ( BlockEdit ) => {
+	return class WrappedBlockEdit extends Component {
```

Comment:
Curious: why enhance the `BlockEdit` component and not let block authors enhance their own `edit` component or a child component of that? The whole `BlockEdit` does not need to rerender on a state change, right? Notices are only shown within the `edit` component of the block, or am I mistaken?

Reply:
Sorry, I missed the update to the component name during the changes, it should have been called OriginalComponent instead of BlockEdit.
withNotices is a generic HOC that can be passed to any react component and it is not being used against BlockEdit we use it inside the blocks edit because as you said notices are only shown within the edit component of the block.